### PR TITLE
Implement gRPC service router.

### PR DESCRIPTION
**Pull request by muzykantov**

### CODE REVIEW

File: core/grpc/compile_protos.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+PROTO_DIR="./proto"
+OUT_DIR="."
+
+mkdir -p $OUT_DIR
+
+protoc --proto_path=$PROTO_DIR \
+       --go_out=$OUT_DIR --go_opt=paths=source_relative \
+       $PROTO_DIR/method_options.proto
+
+echo "Protobuf files compiled successfully!"
```
Comment (pfi79) [inline core/grpc/compile_protos.sh:12]: Why sh-file? Can't you do it without sh-file? Can't you do it some other way?

**Comment by pfi79** [comment]
Please consider creating an integration test with your changes.

File: test/unit/service/proto/balance_service.proto [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+message Address {
+    string address = 1 [(validate.rules).string = {pattern: "^[1-9A-HJ-NP-Za-km-z]+$"}];
+}
+
+// BigInt represents a large integer.
+message BigInt {
+    string value = 1 [(validate.rules).string = {pattern: "^[0-9]+$"}]; // Value must be a number.
+}
+
+// Balance represents a balance with a suffix and value.
```
Comment (zemtsov) [inline test/unit/service/proto/balance_service.proto:21]: suffix or prefix?

File: core/grpc/grpcctx/context.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package grpcctx
```
Comment (vitiko) [inline core/grpc/grpcctx/context.go:1]: may be just ctx

**Pull request closed without merging**